packages/interactive_media_ads/lib/src/platform_interface/platform_companion_ad.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright 2013 The Flutter Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// An object that holds data corresponding to the companion Ad.
+base class PlatformCompanionAd {
+  /// Constructs a [PlatformCompanionAd].
+  PlatformCompanionAd({
+    required this.apiFramework,
+    required this.height,
+    required this.resourceValue,
+    required this.width,
+  });
+
+  /// The API needed to execute this ad, or null if unavailable.
+  final String? apiFramework;
+
+  /// The height of the companion in pixels.
+  final int? height;
+
+  /// The URL for the static resource of this companion.
+  final String? resourceValue;
+
+  /// The width of the companion in pixels.
+  final int? width;
```

Comment:
Height and width should probably be together in the member list and the constructor params.